src/driver/sdfmt.d

```diff
@@ -55,7 +57,23 @@ int main(string[] args) {
 		conf.buildLocalConfig("sdfmt", context, filename);
 
 		import source.location;
-		auto base = context.registerFile(Location.init, filename, "");
+		Position base;
+		if (filename == "-") {
+			// read from stdin
+			import std.array, std.stdio;
+			const stdinData = stdin.byChunk(4096).join;
+
+			import source.util.utf8;
+			const sanitizedData = convertToUTF8(stdinData);
+
+			const fakeFilename =
+				assumeFilename.length ? assumeFilename : "stdin";
```

Comment:
shouldn't "stdin" be changed to something like "." such that config file lookup starts from cwd ?

Reply:
I assumed that's exactly what it'd do, treating it like a local `stdin.d` file in cwd (but having a nicer `stdin` 'name' in potential error msges etc.). But totally untested. ;)